src/modules/categories/categories.controller.ts

```diff
@@ -0,0 +1,20 @@
+import { Controller, Get, Post, Body } from '@nestjs/common';
+import { CategoriesService } from './categories.service';
+
+/**
+ * Controller for managing category-related endpoints.
+ */
+@Controller('categories')
+export class CategoriesController {
+  constructor(private categoriesService: CategoriesService) {}
+
+  @Get()
+  findAll() {
+    return this.categoriesService.findAll();
```

Comment:
Create a DTO response file in `./modules/categories/responses/list-categories.response.ts`. It should be structured like this:

```tsx
export class ListCategoriesResponse<T extends BaseResponse> extends JsonResponse<T> {
  @Expose({ name: 'some_property' })
  someProperty: string; // object.some_property will be converted to object.someProperty
}
```
This structure ensures that the DTO exposes only the necessary properties to the client, and using the `@Expose()` decorator with the class-transformer library allows for camelCase conversion of keys when sending the response.

Here’s how the DTO works with your service in the findAll() method:
```tsx
  @Get()
  async findAll() {
    const categories = await this.categoriesService.findAll();
    return new ListCategoriesResponse().json({ data: categories });
  }
```